test/e2e/reconciler/locationworkspace/synctarget_test.go

```diff
@@ -116,7 +116,7 @@ func TestSyncTargetExport(t *testing.T) {
 		}
 
 		if syncTarget.Status.SyncedResources[0].Resource != "cowboys" ||
-			syncTarget.Status.SyncedResources[0].State != workloadv1alpha1.ResourceSchemaIncompatibleState {
+			syncTarget.Status.SyncedResources[0].State != workloadv1alpha1.ResourceSchemaPendingState {
```

Comment:
Since we're not testing the incompatible state anymore, maybe an additional resource could be added that would be incompatible.
This could be done by:
- creating an additional APIExport with 1 APIResourceSchema in a workspace
- adding the same CRD in the physical cluster (by `WithDownstreamPreparation` when creating the SyncerFixture), but with a small field of the CRD schema changed (for example changing string to int)
- checking that this reosurce on the Synctarget is in the Incompatible state

Reply:
https://github.com/kcp-dev/kcp/blob/main/pkg/reconciler/workload/synctargetexports/synctargetcompatible_reconcile_test.go#L119

I think the test case is already there!
@davidfestal

Reply:
@davidfestal is this needs to be part of same PR?